components/tidb_query/src/rpn_expr/impl_math.rs

```diff
@@ -343,6 +343,33 @@ pub fn atan_2_args(arg0: &Option<Real>, arg1: &Option<Real>) -> Result<Option<Re
     })
 }
 
+#[inline]
+#[rpn_fn]
+pub fn conv(
+    n: &Option<Bytes>,
+    from_base: &Option<Int>,
+    to_base: &Option<Int>,
+) -> Result<Option<Bytes>> {
+    use crate::expr_util::conv::*;
+    if let (Some(n), Some(from_base), Some(to_base)) = (n, from_base, to_base) {
+        let from_base = IntWithSign::from_int(*from_base);
```

Comment:
Can we put the inner logic to the shared code base as well? Then you can also change the non-vectorized implementation to use it, which only a few lines of code.

Reply:
I move the main logic to `conv_internal`. Should I clean the non-vectorized implementation in this PR or create another one after this be merged?